lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -11710,26 +11712,38 @@ void Emit(ParseNode* pnode, ByteCodeGenerator* byteCodeGenerator, FuncInfo* func
     // otherwise. (In other words, the "truth" of the right-hand expression is never tested.)
     // PTNODE(knopLogOr      , "||"        ,None    ,Bin  ,fnopBin)
     case knopLogOr:
+    case knopAsgLogOr:
     {
         STARTSTATEMENET_IFTOPLEVEL(isTopLevel, pnode);
         Js::ByteCodeLabel doneLabel = byteCodeGenerator->Writer()->DefineLabel();
         // We use a single dest here for the whole generating boolean expr, because we were poorly
         // optimizing the previous version where we had a dest for each level
-        funcInfo->AcquireLoc(pnode);
+        const Js::RegSlot result_location = funcInfo->AcquireLoc(pnode);
```

Comment:
NIT: I don't think we use `const` for any other RegSlot temps, so to be consistent please drop it here. Same point for other uses below.

Reply:
In fact there's no need to say result_location here; could revert this edit and use `pnode->location` below, except, you're assigning it to itself by the looks of things with your call below this may be where it's going wrong.

What is going wrong exactly?